rules/sort-imports.ts

```diff
@@ -214,10 +217,19 @@ export default createEslintRule<Options<string[]>, MESSAGE_ID>({
       /* Avoid matching on named imports without specifiers */
       !/\}\s*from\s+/u.test(sourceCode.getText(node))
 
+    let styleExtensions = new Set([
+      '.less',
+      '.scss',
+      '.sass',
+      '.styl',
+      '.pcss',
+      '.css',
+      '.sss',
+    ])
     let isStyle = (value: string): boolean => {
       let [cleanedValue] = value.split('?')
-      return ['.less', '.scss', '.sass', '.styl', '.pcss', '.css', '.sss'].some(
-        extension => cleanedValue.endsWith(extension),
+      return [...styleExtensions].some(extension =>
```

Comment:
[perf] Complexity is not changed here: This is still `O(n)` complexity (`n` is the size of `styleExtensions`):

`styleExtensions` being a `Set` has no benefit here, as the initial array has no duplicate, and the set is converted back into an `Array` with `[...styleExtensions]`.

I think that simply doing

```ts
    let styleExtensions = [
      '.less',
      '.scss',
      '.sass',
      '.styl',
      '.pcss',
      '.css',
      '.sss',
    ]
```

will be the optimal.

Reply:
My bad. Fixed. Thanks!